deepchem/molnet/load_function/kt_dataset.py

```diff
@@ -0,0 +1,239 @@
+from deepchem.molnet.defaults import get_defaults
+
+from typing import List, Tuple, Dict, Optional
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_DIR = deepchem.utils.get_data_dir()
+KTDATASET_URL = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/KTparameterDataset.zip"
+KTDATASET_CSV_URL = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/KTparameterDataset.csv"
+
+# dict of accepted featurizers for this dataset
+# modify the returned dicts for your dataset
+DEFAULT_FEATURIZERS = get_defaults("feat")
+
+# Names of supported featurizers
+# ktdataset_featurizers = ["RawFeaturizer", "SmilesToImage", "SmilesToSeq", "WeaveFeaturizer"]
+ktdataset_featurizers = ['ECFP','GraphConv','Weave','Raw','AdjacencyConv']
+DEFAULT_FEATURIZERS = {k: DEFAULT_FEATURIZERS[k] for k in ktdataset_featurizers}
+
+# dict of accepted transformers
+DEFAULT_TRANSFORMERS = get_defaults("trans")
+
+# dict of accepted splitters
+DEFAULT_SPLITTERS = get_defaults("splits")
+
+# names of supported splitters
+ktdataset_splitters = [
+    'index', 'random', 'scaffold', 'butina', 'task', 'stratified'
+]
+DEFAULT_SPLITTERS = {k: DEFAULT_SPLITTERS[k] for k in ktdataset_splitters}
+
+
+def load_kt_dataset(
+    featurizer: Featurizer = DEFAULT_FEATURIZERS['RawFeaturizer'],
+    transformers: List[Transformer] = [
+        DEFAULT_TRANSFORMERS['PowerTransformer']
+    ],
+    splitter: Splitter = DEFAULT_SPLITTERS['RandomSplitter'],
+    reload: bool = True,
+    data_dir: Optional[str] = None,
+    save_dir: Optional[str] = None,
+    featurizer_kwargs: Dict[str, object] = {},
+    splitter_kwargs: Dict[str, object] = {},
+    transformer_kwargs: Dict[str, Dict[str, object]] = {},
+    **kwargs) -> Tuple[List, Tuple, List]:
+  """
+  Load ktdataset.
+
+  This has been modified from the template for adding a function to load a dataset from
+  MoleculeNet. Global variable URL strings, default parameters,
+  default featurizers, transformers, and splitters, and variable names have been modified as
+  needed. All available featurizers, transformers, and
+  splitters are in the `DEFAULTS_X` global variables.
+
+  If `reload = True` and `data_dir` (`save_dir`) is specified, the loader
+  will attempt to load the raw dataset (featurized dataset) from disk.
+  Otherwise, the dataset will be downloaded from the DeepChem AWS bucket.
+
+  The dataset will be featurized with `featurizer` and separated into
+  train/val/test sets according to `splitter`. Some transformers (e.g.
+  `NormalizationTransformer`) must be initialized with a dataset. 
+  Set up kwargs to enable these transformations. Additional kwargs may
+  be given for specific featurizers, transformers, and splitters.
+
+  The load function must be modified with the appropriate DataLoaders
+  for all supported featurizers for your dataset.
+
+  Please refer to the MoleculeNet documentation for further information
+  https://deepchem.readthedocs.io/en/latest/moleculenet.html.
```

Comment:
You can delete these notes and add relevant info for your dataset. In particular it would be helpful to include information about the size of the dataset, and any previous benchmarking studies that give an idea about what sort of performance should be expected.